src/detect-parse.c

```diff
@@ -818,17 +818,17 @@ static int SigParseOptions(DetectEngineCtx *de_ctx, Signature *s, char *optstr,
         setup_ret = st->Setup(de_ctx, s, NULL);
     }
     if (setup_ret < 0) {
-        SCLogDebug("\"%s\" failed to setup", st->name);
-
         /* handle 'silent' error case */
         if (setup_ret == -2) {
+            SCLogDebug("\"%s\" failed to setup", st->name);
             enum DetectKeywordId idx = SigTableGetIndex(st);
             if (de_ctx->sm_types_silent_error[idx] == false) {
                 de_ctx->sm_types_silent_error[idx] = true;
                 return -1;
             }
             return -2;
         }
+        SCLogWarning(SC_ERR_INVALID_RULE_ARGUMENT, "\"%s\" failed to setup", st->name);
```

Comment:
this is not very helpful wrt user feedback. It doesn't give enough context about the issue.

Also, this should be an error, not a warning, as the rule will fail to parse

Reply:
Ok for error vs warning.

> this is not very helpful wrt user feedback. It doesn't give enough context about the issue.

I am not sure what you mean.
This brings more context about the issue than what is currently being done, as suggested in https://github.com/OISF/suricata/pull/7611#discussion_r929672047

When a signature fails to parse, we know already which one, we may have a log error from the keyword.
With this PR, we will know every time which keyword caused the failure.

Am I missing something ?

Reply:
It just adds another line of output, which has some useful info (keyword name), but makes errors even more verbose and noisy. I would like a solution where we don't add more error/warning lines for a single issue. I kind of dislike the current minimum of 2 lines (keyword issues specific error, then parser issues a generic one which includes the sig). Ideally we go from 2 to 1, not from 2 to 3 :)

Reply:
Ok, got it

The solution will not be a one-liner then :-/